### PR TITLE
Filter out failed test-perf runs.

### PR DESCRIPTION
**Pull request by asajeffrey**
Google Data Studio is a lot happier if the data that's driving it is already filtered. This PR filters out any failed test runs from the CSV file generated by test-perf.

---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes do not require tests because this is test infrastructure

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19523)

### CODE REVIEW

File: etc/ci/performance/runner.py [hunk 1]
```diff
@@ -281,10 +281,12 @@ def save_result_csv(results, filename, manifest, expected_runs, base):
         'unloadEventStart',
     ]
 
+    successes = list(filter(lambda x: x['domComplete'] != -1, results))
```
Comment (aneeshusa) [inline etc/ci/performance/runner.py]: List comprehensions are more idiomatic in Python:
```
successes = [r for r in results if r['domComplete'] != -1]
```

Comment (asajeffrey) [reply etc/ci/performance/runner.py]: Fixed. There's quite a few more of those :/ Time to file an E-Easy issue?

**Comment by avadacatavra** [comment]
@asajeffrey looks good--can you squash please?

**Comment by asajeffrey** [comment]
Will do when I'm back in front of a computer.

**Comment by avadacatavra** [comment]
@asajeffrey status on this?

**Comment by asajeffrey** [comment]
@avadacatavra er the status is Xmas holiday. I squashed.

@bors-servo r=avadacatavra

**Comment by asajeffrey** [comment]
@bors-servo retry
* #19495

**Pull request merged**